src/parser.ts

```diff
@@ -95,16 +98,23 @@ function _isASTNode(node: unknown): node is ASTNode {
     return false
   }
 
-  const nodeAsAny: any = node
+  const nodeAsAny = node as ASTNode
```

Comment:
Line [107](https://github.com/solidity-parser/parser/pull/91/files#diff-abe2818e9c29aadf84401cb7ef8286c7cbce9253eba09f6fa52e22ca4e57b17eR107)

`return astNodeTypes.includes(nodeAsAny.type)`

expects `nodeAsAny.type` to be an `astNodeTypes` (might work with `string` but did not test that option)